luigi/target.py

```diff
@@ -260,10 +260,11 @@ class AtomicLocalFile(io.BufferedWriter):
     :class:`luigi.file.LocalTarget` for example
     """
 
-    def __init__(self, path):
+    def __init__(self, path, mode='w'):
         self.__tmp_path = self.generate_tmp_path(path)
         self.path = path
-        super(AtomicLocalFile, self).__init__(io.FileIO(self.__tmp_path, 'w'))
+        self.mode = mode
+        super(AtomicLocalFile, self).__init__(io.FileIO(self.__tmp_path, mode))
```

Comment:
This should be `super(AtomicLocalFile, self).__init__(io.FileIO(self.__tmp_path, self.mode))` since you define `self.mode = mode`.